### PR TITLE
feat(curriculum): add rock paper scissors workshop

### PR DESCRIPTION
**Pull request by jdwilkin4**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/workshop-rps-game/66cf13fe7619e9a1d57c7b5e.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+dashedName: step-2
+---
+
+# --description--
+
+The next step is to build out the functionality that will generate a random choice for the computer.
+
+Create a function called `getRandomComputerResult` that returns a random choice from the `options` array.
+
+*Hint*: Remember you learned about `Math.random()` and `Math.floor()` in the earlier JavaScript  fundamentals section.
```
Comment (Sembauke) [inline curriculum/challenges/english/25-front-end-development/workshop-rps-game/66cf13fe7619e9a1d57c7b5e.md:14]:
```suggestion
*Hint*: Remember you learned about `Math.random()` and `Math.floor()` in the earlier JavaScript fundamentals section. 
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Review by Sembauke — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/workshop-rps-game/66cf35a75f891ab4f4e5497b.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+# --description--
+
+If you try to play the game, you will see that you can play for an infinite amount of rounds. But the rules state that the first one to three points wins.
+
+Inside your `showResults` function, you will need to check if the player or computer has reached three points. If either has reached three points, you should display a message indicating the winner. 
+
+For example, if the player has won the game, then the `winnerMsgElement` should be updated to `"Player has won the game!"`. If the computer has won the game, then the `winnerMsgElement` should be updated to `"Computer has won the game!"`.
+
+If there is a winner, you will want to show the `resetGameBtn` button and hide the `optionsContainer` so the player can play again.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/workshop-rps-game/66cf35a75f891ab4f4e5497b.md:16]:
```suggestion
If you try to play the game, you will see that you can play for an infinite amount of rounds. But the rules state that the first one to three points wins. You want to check if there's a winner, and display a message.

In your `showResults` function, if the player has reached three points, update the `winnerMsgElement` to `"Player has won the game!"`. If the computer has reached three points, update the `winnerMsgElement` to `"Computer has won the game!"`.

If there is a winner, show the `resetGameBtn` button by settings it's `display` to `block` and hide the `optionsContainer` by setting it's `display` to `none`.
```

I struggled with this one a bit, the instructions weren't all that clear to me - I came up with a suggestion.

**Review by moT01 — CHANGES_REQUESTED** [review]
This is pretty awesome @jdwilkin4 🎉 I have number of suggestions...

I would like to see a little more directing users to play with the app. There’s a little of that in there, which is nice - but at the beginning or end of the instructions on some more steps, you could add some more… “If you try to play the game now, [describe functionality]” or “After you finish, test your game, you will see [functionality]” Just a suggestion, no need to add any more - like I said, there is some of that in there already.

No need to do this one either, just a suggestion again. It feels like I'm thrown into the middle of the project a bit. On step 1, you could maybe describe the boilerplate given to them and tell them to look at it - and what functionality they will add. Or maybe even have a few steps at the beginning to have them finish up the HTML so they can see the buttons and things they are adding. I think what's bothering me is that I am not familiar with the boilerplate and don't have an overview of what code I have or quite what I am going to build. I'm not sure what the solution to that is. Wonder if we should add the `demoType: onLoad` here? was that left out intentionally? It might help a bit.

step5 - The `getRoundResults` function is started for me. We could add another step and let the users create that?

Step7 - didn’t pass with this code:
```
  function showResults(userOption) {
  playerScoreSpanElement.innerText = playerScore;
  computerScoreSpanElement.innerText = computerScore;
  roundResultsMsg.innerText = getRoundResults(userOption);
}
```
It passed when I moved the last line (the roundResults…) to the top of the function. The instructions should change to specify the order these should be in, or the tests should change to allow any order.

Step 9 - maybe change the wording. Instead of “...you should call the showResults function...” maybe “...you should run a function that calls the showResults function…”

Not all of the suggestions are necessary, I think steps 7, 9, and 11 should probably get some adjustments. Let me know if you have any thoughts on any of it.

**Comment by jdwilkin4** [comment]
> I would like to see a little more directing users to play with the app.

Yeah I agree. I can update that 👍 

> Wonder if we should add the demoType: onLoad here? was that left out intentionally? It might help a bit.

I wasn't sure if that was only for labs but I can add that in

> On step 1, you could maybe describe the boilerplate given to them and tell them to look at it - and what functionality they will add.

We could do similar to the current pyramid geneartor project where we give a brief intro into what they are going to do

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-introductory-javascript-by-building-a-pyramid-generator/step-1

We could explain what the boilerplate is there and encourage them to check it out before proceeding to the next step.

We could also do this for all DOM projects where the HTML and CSS is provided for them

**Comment by moT01** [comment]
> I wasn't sure if that was only for labs but I can add that in

Yea, the labs should have a `demoType` of `onClick` I believe - that will show the button to open the preview. And step-1 of the workshops should have a `demoType` of `onLoad` I believe - which opens the preview on that step. [See step-1 of the current quiz project](https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/main/curriculum/challenges/english/14-responsive-web-design-22/learn-accessibility-by-building-a-quiz/614ccc21ea91ef1736b9b578.md) - and [see it in action](https://www.freecodecamp.dev/learn/2022/responsive-web-design/learn-accessibility-by-building-a-quiz/step-1)

**Comment by jdwilkin4** [comment]
I will work on applying the other changes tomorrow 👍

**Comment by jdwilkin4** [comment]
> step5 - The getRoundResults function is started for me. We could add another step and let the users create that?

The issue with that is there is not good way to test that they created variable and assigned it the result of called the `getRoundResults`. We would have to use regex tests here, which I don't think we should do here

```js
function getRoundResults(userOption) {
  const computerResult = getRandomComputerResult();
 
}
```

even if we try to update step 6 to have them create the function and mention that they should create a variable of their choice and assign that function call, the directions are already long enough as is. So my concern of adding another sentence of two will make it to long and crowded.

**Review by moT01 — APPROVED** [review]
LGTM @jdwilkin4 🎉 It's pretty fun 👍

I still think I would prefer to see step 6 broken up into two steps and use a regex test there if needed.

Also, it feels a little... strange that step 1 doesn't have a task to complete. It would be nice if we could bake a task in with the description - but it's pretty long already, so I'm fine with it.

I'm going to go ahead and merge this. Those feel like preferences - we can make adjustments on another PR if we really want to.

**Pull request merged**